infrastructure/cloudconnect/gcp-vpn-buildconf.py

```diff
@@ -0,0 +1,139 @@
+        'peerBgpIpAddress': 'vpn_gateway/tunnel_inside_address/ip_address'
+    },
+    'static_tunnel': {
+        'peerIp': 'vpn_gateway/tunnel_outside_address/ip_address',
+        'sharedSecret': 'ike/pre_shared_key',
+    }
+}
+
+BGP_CONFIG = """
+imports: 
+  - path: gcp-vpn.jinja
+        
+resources: 
+  - name: vpn
+    type: gcp-vpn.jinja
+    properties: 
+      network: {{ network }}
+      region: {{ region }} 
+      address: {{ address }}
+      asn: {{ asn }}
+      tunnels:
+      {%- for i in tunnels %}
+        - peerIp: {{ i['peerIp'] }}
+          secret: {{ i['sharedSecret'] }}
+          peerAsn: {{ i['peerAsn'] }}
+          bgpIpAddress: {{ i['bgpIpAddress'] }}
+          peerBgpIpAddress: {{ i['peerBgpIpAddress'] }}
+      {%- endfor %}
+"""
+
+STATIC_CONFIG = """
+imports: 
+  - path: gcp-vpn.jinja
+        
+resources: 
+  - name: vpn
+    type: gcp-vpn.jinja
+    properties: 
+      network: {{ network }}
+      region: {{ region }} 
+      address: {{ address }}
+      tunnels:
+      {%- for i in tunnels %}
+        - peerIp: {{ i['peerIp'] }}
+          secret: {{ i['sharedSecret'] }}
+          localTrafficSelector:
+          {%- for j in i['local_traffic_selector'] %}
+            - {{ j }}
+          {%- endfor %}
+          remoteTrafficSelector:
+          {%- for j in i['remote_traffic_selector'] %}
+            - {{ j }}
+          {%- endfor %}
+      {%- endfor %}
+"""
+
+def main():
+    ''' main '''
+
+    parser = argparse.ArgumentParser(description="""
+        Reads a CustomerGatewayConfiguration(XML) from stdin, and outputs a yaml file to use with gcp-vpn.jinja.""", usage="""
+        aws ec2 describe-vpn-connections --filter Name=vpn-connection-id,Values=vpn-67c00420 --query VpnConnections[0].CustomerGatewayConfiguration --output text | ./gcp-vpn-buildconf.py --network https://www.googleapis.com/compute/v1/projects/xpn-host/global/networks/vpc >> gcp-vpn.yaml""")
+
+    parser.add_argument('--region', default='us-east1')
+    parser.add_argument('--network', required=True,
+                        help='Fully-qualified network url')
+    parser.add_argument('--local-traffic-selector', nargs='+')
+    parser.add_argument('--remote-traffic-selector', nargs='+')
+    args = parser.parse_args()
+
+    try:
+        tree = ET.parse(sys.stdin)
+    except ValueError:
+        print("InValid xml input. Verify aws cli command includes \"--query \
+            VpnConnections[0].CustomerGatewayConfiguration \
+            --output text\"", file=sys.stderr)
+        sys.exit(1)
+
+    params = {
+        'region': args.region,
+        'network': args.network,
+        'tunnels': []
+    }
+
+    for i in tree.getroot().findall('ipsec_tunnel'):
+        params['address'] = i.find(PATHS['address']).text
+        if i.find(PATHS['asn']) is None:
+            config = STATIC_CONFIG
+            tunnel = {k: i.find(v).text for k, v in \
```

Comment:
tunnel = {k: i.find(v).text for k, v in 
                        PATHS['static_tunnel'].items()}